agent/lib/kontena/helpers/wait_helper.rb

```diff
@@ -0,0 +1,32 @@
+require_relative '../logging'
+
+module Kontena
+  module Helpers
+    module WaitHelper
+      include Kontena::Logging
+
+      def wait(timeout = 300, message = nil, &block)
+        wait_until = Time.now.to_f + timeout
+        loop do
+          raise ArgumentError, 'no block given' unless block_given?
+          value = yield
+          return value if value || !still_waiting?(wait_until)
+          debug message if message
+          sleep 0.5
+        end
+      end
+
+      def still_waiting?(wait_until)
```

Comment:
Let's rename this to `_still_waiting?` or similar if this is considered as a "private" method that should not pollute including class methods.